src/compile/data/optimizers.ts

```diff
@@ -398,3 +399,11 @@ export class MergeBins extends BottomUpOptimizer {
     return this.flags;
   }
 }
+
+/**
+ * Wraps fieldIntersection so that it can be called with parent and child nodes directly
+ * and check whether or not the parent node is a pivot transform
+ */
+export function nodeFieldIntersection(parent: DataFlowNode, child: DataFlowNode): boolean {
+  return parent instanceof PivotTransformNode || fieldIntersection(parent.producedFields(), child.producedFields());
```

Comment:
This should fix the issue where histogram examples are changing

```suggestion
  return parent instanceof PivotTransformNode || fieldIntersection(parent.producedFields(), child.dependentFields());
```

Reply:
Missed this fix in the rebase somehow, but this suggestion should fix it

```suggestion
  return parent instanceof PivotTransformNode || fieldIntersection(parent.producedFields(), child.dependentFields());
```

Reply:
Ahh, okay. I was wondering. I think I will rewrite the logic a bit. 

I'm thinking that the intersectin check should return true if the produced or dependent fields is undefined since this means we don't have information to be confident that there is no intersection.